### PR TITLE
Improve mock updating

### LINKED ISSUE #127: [FEATURE] Simplify process to add mocks of new redcaps or improve docs
# Feature Request Description
Adding a new redcap source to mock is currently too complicated. You need to:
1. Assign a fake token in `inst/misc/fake_credentials.csv`
2. Add the token name to `inst/httptest/redact.R` so it gets redacted
3. Add a `Sys.getenv()` call to `tests/testthat/fixtures/create-httptest-mock.R` to get the token before creating the mocks
4. Add a `get_fake_credentials()` call to `tests/testthat/helper.R` to load the fake token for tests

# Proposed Solution
- Step 2 can be eliminated by having `redact.R` load all fake tokens from `fake_credentials.csv`
- Steps 3 and 4 can be eliminated by having `create-httptest-mock.R` and `helper.R` load a single token list to reference later:

helper.R:
```r
creds <- get_credentials(fake = TRUE)

...

read_redcap(creds$redcap_uri, creds$classic_token)
```

create-httptest-mock.R:
```r
creds <- get_credentials()

...

read_redcap(creds$redcap_uri, creds$classic_token)
```

### Checklist

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [x] Developers are assigned to the issue
- [x] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by ezraporter**
# Description
This PR improves our process for adding mocks from new REDCaps and cleans up documentation. In doing this I made a slight change to how we reference credentials in our test suite. Before we'd have:

```r
classic_token <- get_fake_credentials("REDCAPTIDIER_CLASSIC_API")

httptest::with_mock_api({
    read_redcap(redcap_uri, classic_token)
})
```

To avoid having to assign values in multiple places I cut out the "classic_token" so now out test suite loads all credentials in a `creds` object and you use it like:

```r
httptest::with_mock_api({
    read_redcap(redcap_uri, creds$REDCAPTIDIER_CLASSIC_API)
})
```

# Proposed Changes 
List changes below in bullet format:

- Replace `get_fake_credentials()` with `get_credentials()` that will load real or fake credentials in a standard format
- Replace instances of using credentials with the new `creds$` syntax described above
- Added [`updating-mocks.md`](https://github.com/CHOP-CGTInformatics/REDCapTidieR/blob/improve-mock-updating/updating-mocks.md) with instructions on adding and updating mocks

I also noticed that since we updated to `lintr 3.0.2` our `expect_lint_free()` test hasn't been working. It warns but doesn't lint anything. I think this is a bug with `lintr` so I'll file an issue there. For now I rolled back the version of `lintr` in our `renv.lock` so the test works locally.

### Issue Addressed
Closes #127

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [x] New/revised functions that update downstream outputs have associated static testing files (`.RDS`) updated under `inst/testdata/create_test_data.R`
- [x] New tests that make API calls use `httptest::with_mock_api` and any new mocks were added to `tests/testthat/fixtures/create_httptest_mocks.R`
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR
- [x] Pre-release package version incremented using `usethis::use_dev_version()`

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

**Comment by ezraporter** [comment]
> I also noticed that since we updated to `lintr 3.0.2` our `expect_lint_free()` test hasn't been working. It warns but doesn't lint anything. I think this is a bug with `lintr` so I'll file an issue there. For now I rolled back the version of `lintr` in our `renv.lock` so the test works locally.

Okay actually confirmed that this is fixed in the development version on github. Probably best to keep it at `3.0.1` for now and bump to `3.0.3` when it's on CRAN.

File: tests/testthat/helper.R [hunk 1]
```diff
@@ -1,31 +1,38 @@
-# This function behaves like Sys.getenv() but returns values from the
-# store of fake REDCap API credentials in inst/misc
+# This function returns a list of named credentials either with `Sys.getenv()` if `fake = FALSE`
+# or from `inst/misc/fake_credentials.csv` if `fake = TRUE`
+#
+# By default all credentials stored in `fake_credentials.csv` are returned
 
-get_fake_credentials <- function(credentials) {
+get_credentials <- function(credentials = NULL, fake = FALSE) {
```
Comment (rsh52) [inline tests/testthat/helper.R:6]: Is it at all worth moving this out to the `R/dev_utils.R` location so you don't have to copy it in two locations? Or is it necessary to have it in both locations for it to work properly?

**Review by rsh52 — APPROVED** [review]
Just one comment, otherwise looks good and thank you for simplifying this!

File: tests/testthat/helper.R (hunk 1 above)
Comment (ezraporter) [reply tests/testthat/helper.R:6]: We could do that but the only catch is it would need to be an exported function to work in the vignette. I think other solutions like referring to it with `:::` upset R CMD Check
